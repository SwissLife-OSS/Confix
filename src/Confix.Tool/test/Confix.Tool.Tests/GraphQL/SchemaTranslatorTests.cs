using System.Text.Json;
using Confix.Tool.Schema;
using Json.Schema;
using Snapshooter.Xunit;

namespace ConfiX.GraphQL;

public sealed class SchemaTranslatorTests
{
    [Fact]
    public void Scalar_DefaultValue_Should_BeTranslatedCorrectly()
    {
        // arrange
        string schema = """
            type Query {
                foo: String @defaultValue(value: "bar")
            }
        """;

        // act
        var result = SchemaHelpers.BuildSchema(schema).ToJsonSchema();

        // assert
        Match(result);
    }

    [Fact]
    public void Object_DefaultValue_Should_BeTranslatedCorrectly()
    {
        // arrange
        string schema = """
            type Query {
                foo: Foo @defaultValue(value: { bar: "baz" })
            }
            type Foo {
                bar: String
            }
        """;

        // act
        var result = SchemaHelpers.BuildSchema(schema).ToJsonSchema();

        // assert
        Match(result);
    }

    private static void Match(JsonSchemaBuilder builder) => Serialize(builder).MatchSnapshot();

    private static string Serialize(JsonSchemaBuilder builder)
        => JsonSerializer.Serialize(builder.Build(),
            new JsonSerializerOptions() { WriteIndented = true });
}
