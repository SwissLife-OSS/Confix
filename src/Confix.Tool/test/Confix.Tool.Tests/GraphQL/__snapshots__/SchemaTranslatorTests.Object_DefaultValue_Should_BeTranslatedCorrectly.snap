{
  "type": "object",
  "properties": {
    "foo": {
      "anyOf": [
        {
          "$ref": "#/$defs/Foo"
        },
        {
          "type": "null"
        }
      ],
      "deprecated": false,
      "default": {
        "bar": "baz"
      }
    }
  },
  "required": [],
  "additionalProperties": false,
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "$defs": {
    "Query": {
      "type": "object",
      "properties": {
        "foo": {
          "anyOf": [
            {
              "$ref": "#/$defs/Foo"
            },
            {
              "type": "null"
            }
          ],
          "deprecated": false,
          "default": {
            "bar": "baz"
          }
        }
      },
      "required": [],
      "additionalProperties": false
    },
    "Foo": {
      "type": "object",
      "properties": {
        "bar": {
          "anyOf": [
            {
              "$ref": "#/$defs/String"
            },
            {
              "type": "null"
            }
          ],
          "deprecated": false
        }
      },
      "required": [],
      "additionalProperties": false
    },
    "String": {
      "type": "string",
      "hasVariable": true,
      "description": "The \u0060String\u0060 scalar type represents textual data, represented as UTF-8 character sequences. The String type is most often used by GraphQL to represent free-form human-readable text."
    },
    "Boolean": {
      "type": "boolean",
      "hasVariable": true,
      "description": "The \u0060Boolean\u0060 scalar type represents \u0060true\u0060 or \u0060false\u0060."
    },
    "JSON": {
      "type": "string",
      "hasVariable": true
    }
  }
}
